docs/src/_reference/plugin-definition-syntax.md

```diff
@@ -0,0 +1,399 @@
+---
+title: Plugin Definition Syntax
+description: YAML syntax for defining plugins for Meltano Hub.
+layout: doc
+weight: 3
+---
+
+This document describes the syntax for defining plugins contributed to Meltano Hub.
+
+## `name`
+
+Required. The name of the plugin. The tuple `(name, variant)` must be unique.
+
+```yaml
+name: airflow
+```
+
+## `variant`
+
+Required. The name of the variant. New variants of existing plugins usually come form forks or re-implementations.
+
+```yaml
+variant: apache
+```
+
+## `namespace`
+
+Required. This is used to configure multiple plugins that are meant to work together.
+
+```yaml
+namespace: airflow
+```
+
+## `label`
+
+A human-readable label for the plugin.
+
+```yaml
+label: Airflow
+```
+
+## `docs`
+
+The URL of the plugin's documentation.
+
+```yaml
+docs: https://docs.meltano.com/guide/orchestration
+```
+
+## `repo`
+
+The URL of the plugin's repository (in GitHub, GitLab, etc.).
+
+```yaml
+repo: https://github.com/apache/airflow
+```
+
+## `capabilities`
+
+Array of capabilities that the plugin supports.
+
+```yaml
```

Comment:
Is this the full array? If not (or even if yes) is there a canonical place we can link to for the most up to date list?

Reply:
AFAICT the closest toa a canonical source for admitted capabilities is the JSON schemas. The only plugin types that declare a `capabilities` enum are [extractors](https://github.com/meltano/hub/blob/44405ce4e3ccc3b3d8630673fc210f058bb5da7a/schemas/common/extractors_specific.schema.json#L9) and [loaders](https://github.com/meltano/hub/blob/44405ce4e3ccc3b3d8630673fc210f058bb5da7a/schemas/common/loaders_specific.schema.json#L9). Perhaps @pnadolny13 can confirm if there's a SSOT for plugin capabilities.

Reply:
@edgarrmondragon not critical to block on this. Something to be aware of as we consider adding more capabilities with the Plugin SDK.